src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -264,10 +264,14 @@ async def _finish_handler_with_unary_response(RPCState rpc_state,
     rpc_state.raise_for_termination()
 
     # Serializes the response message
-    cdef bytes response_raw = serialize(
-        response_serializer,
-        response_message,
-    )
+    cdef bytes response_raw
+    if rpc_state.status_code == StatusCode.ok:
+        response_raw = serialize(
+            response_serializer,
+            response_message,
+        )
+    else:
+        response_raw = b''
```

Comment:
I think this is something that we should document about serialization, since users are allowed to pass in a serializer that takes `None` to be a meaningful value. Since it looks like the documentation is currently just a sentence repeated for each of the arities, maybe we can add a new section explicitly about serializers and deserializers?

Reply:
The documentation change is created at https://github.com/grpc/grpc/pull/22677. `response_raw` is passed down to Core, so it must be an object with `bytes` type.

This PR handles the case that the handler returns an implicit error. It should not impact users whose serializer accepts `None` value.

Reply:
[A non-OK status does not mean that you cannot set a response message, right?](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#responses). What if the user has defined their own serializer (that has a valid serialization for `None`) and in addition to setting an error code, returns a response of `None`?

Reply:
Yes. According to the document, applications can't send response message with non-OK status. If the error code is set to non-OK, I think we should consider the RPC is aborted and the response should be discarded (the `None` value is discarded).